src/utils.rs

```diff
@@ -124,4 +132,17 @@ mod tests {
         assert!(utils::subdomain_match("*", "test1.test2.example.com"));
         assert!(utils::subdomain_match("example.com", "example.com"));
     }
+
+    #[test]
+    fn path_match() {
+        assert!(utils::path_match("/cats", "/cats"));
+        assert!(utils::path_match("/", "/"));
+        assert!(utils::path_match("/*", "/"));
+        assert!(utils::path_match("/*", "/cats/something"));
+        assert!(utils::path_match("/cats/*", "/cats/"));
+        assert!(utils::path_match("/cats/*", "/cats/something"));
+        assert!(!utils::path_match("/cats/*", "/cats"));
+        assert!(utils::path_match("/cats*", "/catsanddogs"));
```

Comment:
I was actually surprised by this… because of [this you said](https://kuadrant.slack.com/archives/C02PB55TERE/p1662736074703149):

>TIL that the prefix path match definition of Gateway API is not what I would expect. https://gateway-api.sigs.k8s.io/v1alpha2/references/spec/#gateway.networking.k8s.io/v1alpha2.HTTPRouteMatch
>/abc matches /abc , /abc/ , /abc/cde, but does not match /abcd :exploding_head:

Reply:
well, kuadrant does not implement segment path prefix matching, just strict prefix matching. 

Should we implement in the same way?

Reply:
That's a good question… that i don't plan on answering on this PR! 
I'd merge just merge that, but I'd like to wrap my head around what it all means moving forward… Tho, we can always do some replacement at "higher" abstractions that'd consider `/abc*` segment path prefix matching (if that's how its called), but as `/abc/*` for that layer of the stack… So I don't feel that the cost of this turning out be being wrong would be huge… So remains a 👍 from me!